### PR TITLE
[master] Warn users about adding windows and linux nodes

### PR DESCRIPTION
**Pull request by rak-phillip**
This displays warning to the user, notifying them that workloads will deploy to both windows and linux nodes. This warning is displayed while registering windows nodes or viewing a collection of nodes for a cluster that contains windows nodes. 

#3921

### CODE REVIEW

File: models/management.cattle.io.cluster.js [hunk 1]
```diff
@@ -294,6 +294,10 @@ export default {
     };
   },
 
+  provisioningCluster() {
+    return this.$getters['all'](CAPI.RANCHER_CLUSTER).find(c => c.name === this.id);
```
Comment (rak-phillip) [inline models/management.cattle.io.cluster.js:298]: @vincent99 what I need to know is if this line will always return what we expect it to. This method was taken from [management.cattle.io.node](https://github.com/rancher/dashboard/blob/43d109b2bdc2672ccc9736d9c7cff74cdb7c6314/models/management.cattle.io.node.js#L104-L106). What we found was that the `id` for `management.cattle.io.cluster` isn't namespaced, so we're not sure if this is a safe way to perform our lookups.

File: assets/translations/en-us.yaml [hunk 2]
```diff
@@ -912,6 +912,7 @@ cluster:
       linuxDetail: Run this command on each of the existing Linux machines you want to register.
       windowsDetail: Run this command in Powershell on each of the existing Windows machines you want to register.
       windowsNotReady: The cluster must be up and running with Linux etcd, control plane, and worker nodes before the registration command for adding Windows workers will display.
+      windowsWarning: Registering Windows and Linux worker nodes will allow workloads to deploy on both Linux and Windows worker nodes.
```
Comment (rak-phillip) [inline assets/translations/en-us.yaml:915]: I still need some official copy for the message that we will want to display.

@vincent99 @nwmac

**Comment by rak-phillip** [comment]
I still need to get a setup where I can actually test with windows nodes, but I think the changes are ready for review overall.

**Comment by rak-phillip** [comment]
### New banner during windows registration
image

### New banner while listing nodes
image

**Review by lvuch — APPROVED** [review]
lgtm

**Review by mantis-toboggan-md — APPROVED** [review]
LGTM; I borrowed a windows compatible setup and peeped the warnings with my own eyes.

**Pull request merged**